var/spack/repos/builtin/packages/intel-oneapi-compilers-classic/package.py

```diff
@@ -42,9 +46,22 @@ def setup_run_environment(self, env):
 
         and from setting CC/CXX/F77/FC
         """
-        bin = join_path(self.spec['intel-oneapi-compilers'].prefix,
-                        'compile', 'linux', 'bin', 'intel64')
-        env.set('CC', join_path(bin, 'icc'))
-        env.set('CXX', join_path(bin, 'icpc'))
-        env.set('F77', join_path(bin, 'ifort'))
-        env.set('FC', join_path(bin, 'ifort'))
+        env.set('CC', join_path(self.oneapi_compiler_prefix.bin, 'icc'))
+        env.set('CXX', join_path(self.oneapi_compiler_prefix.bin, 'icpc'))
+        env.set('F77', join_path(self.oneapi_compiler_prefix.bin, 'ifort'))
+        env.set('FC', join_path(self.oneapi_compiler_prefix.bin, 'ifort'))
+
+    def install(self, spec, prefix):
+        # We create a physical directory for binaries because they are more likely
+        # to be affected by being symlinked
+        mkdirp(prefix.bin)
```

Comment:
Should this be?
```
os.symlink(self.oneapi_compiler_prefix.bin.intel64, prefix.bin)
```

Reply:
I'm always scared of linking executable dirs because they like to care about where they come from, but if you say it works in this case I'll go with that. And I definitely need the `intel64` dir in there.